src/text-link.tsx

```diff
@@ -64,10 +64,10 @@ const TextLink = ({
         <BaseTouchable
             {...props}
             stopPropagation
-            as={props.onPress ? 'a' : undefined}
```

Comment:
Using <a/> without setting href will cause the element to not be focusable. This is how HTML works and it's expected.

It makes more sense for the TextLink to be rendered as a button if we are passing onPress instead of href